allennlp/modules/transformer/t5.py

```diff
@@ -806,6 +835,35 @@ def __init__(
 
         self.beam_search = beam_search.construct(end_index=self.eos_token_id)
 
+    def resize_token_embeddings(
+        self, new_size: int, *, init_fn: Callable = torch.nn.init.normal_
+    ) -> None:
+        self.encoder.resize_token_embeddings(new_size, init_fn=init_fn)
+        # If encoder and decoder share embeddings, this is a no-op the second time.
+        self.decoder.resize_token_embeddings(new_size, init_fn=init_fn)
+
+        # resize lm head
+        old_size = self.lm_head.out_features
+        if old_size == new_size:
```

Comment:
Minor thing: Maybe we can do this check first thing? It'll avoid the 2 empty calls.

Reply:
Or perhaps not? We want to resize the embedding dim and the output dim to the same `new_size`. Will the old `embedding_dim` and the old `lm_head.out_features` be different?

Reply:
It's a bit of defensive programming. It should never happen, but if the three places where the size of the token embedding matters get out of sync, this call will sync them all back up.